File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxWidth = 0;
    let maxHeight = 0;
    
    for (const size of sizes) {
        let [width, height] = size;
        if (width < height) {
            const tmp = width;
            width = height;
            height = tmp;
        }
        maxWidth = Math.max(width, maxWidth);
        maxHeight = Math.max(height, maxHeight);
    }
    
    return maxWidth * maxHeight;
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_width = 0
    max_height = 0

    for size in sizes:
        width, height = size

        if width < height:
            width, height = height, width

        max_width = max(max_width, width)
        max_height = max(max_height, height)

    wallet_size = max_width * max_height

    return wallet_size